test/draw_test.py

```diff
@@ -1,7 +1,204 @@
+    depths = [16, 32]
+    surfaces = []
+    for size in sizes:
+        for depth in depths:
+            # Create each possible surface type
+            surface_default = pygame.display.set_mode(size, 0, depth)
+            surface_default_SRCALPHA = pygame.Surface(size, SRCALPHA, depth)
+            surface_alpha = surface_default.convert_alpha()
+            surface_alpha_SRCALPHA = surface_default_SRCALPHA.convert_alpha()
+
+            surfaces.extend([surface_default, surface_alpha,
+                             surface_alpha_SRCALPHA, surface_alpha_SRCALPHA])
+
+    return colors, surfaces
+
+
+class DrawLineTest(unittest.TestCase):
+    """Class for testing line(), aaline(), lines() and aalines().
+    """
+
+    def test_line_color(self):
+        """|tags: ignore|
+
+        Checks if the line drawn with line_is_color() is the correct color.
+        """
+
+        def line_is_color(surface, color, draw_line):
+            """
+            Returns True if draw_line is drawn with the correct color on the
+            given surface.
+            """
+            draw_line(surface, color, (0, 0), (1, 0))
+            return surface.get_at((0, 0)) == color
+
+        for draw_line in [draw.line, draw.aaline]:
+            colors, surfaces = lines_set_up()
+            for surface in surfaces:
+                for color in colors:
+                    self.assertTrue(line_is_color(surface, color, draw_line))
+
+    def test_line_gaps(self):
+        """|tags: ignore|
+
+        Tests if the line drawn with line_has_gaps() contains any gaps.
+
+        See: #512
+        """
+
+        def line_has_gaps(surface, draw_line):
+            """Returns True if the line drawn on the surface contains gaps.
+            """
+            width = surface.get_width()
+            color = (255, 255, 255)
+
+            draw_line(surface, color, (0, 0), (width - 1, 0))
+
+            colors = [surface.get_at((width, 0)) for width in range(width)]
+
+            return len(colors) == colors.count(color)
+
+        for draw_line in [draw.line, draw.aaline]:
+            _, surfaces = lines_set_up()
+            for surface in surfaces:
+                self.assertTrue(line_has_gaps(surface, draw_line))
+
+    def test_lines_color(self):
+        """|tags: ignore|
+
+        Tests if the lines drawn with lines_are_color() are the correct color.
+        """
+        def lines_are_color(surface, color, draw_lines):
+            """Draws (aa)lines around the border of the given surface and
+            checks if all borders of the surface only contain the given color.
+            """
+            width = surface.get_width()
+            height = surface.get_height()
+            points = [(0, 0), (width - 1, 0), (width - 1, height - 1),
+                      (0, height - 1)]
+
+            draw_lines(surface, color, True, points)
+
+            borders = get_border_values(surface, width, height)
+            return [color in border for border in borders]
```

Comment:
Here also little question : `color in border` tests that the color appears at least one time on the border ;  if I use *any* (he is a good friend of *all*), it is equivalent to : 

```python
return  [any(c == color for c in border) for border in borders]

# but shouldn't it be ?
return  [all(c == color for c in border) for border in borders]
```

Reply:
I see what you mean, I will also update that.